metrics/metrics_cli.go

```diff
@@ -0,0 +1,230 @@
+	http.HandleFunc("/", healthCheckHandler)
+	http.Handle("/metrics", promhttp.HandlerFor(promRegistry, promhttp.HandlerOpts{}))
+
+	<-done
+	log.Print("Server Stopped")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer func() {
+		cancel()
+	}()
+
+	if err := srv.Shutdown(ctx); err != nil {
+		log.Fatalf("Server Shutdown Failed:%+v", err)
+	}
+	log.Print("Server Exited")
+
+}
+
+func createCounter(count int, mc metricCollector) {
+	mc.registerCounter(count)
+	updateLoop(mc.updateCounter, mc.interval)
+}
+
+func createGauge(count int, mc metricCollector) {
+	mc.registerGauge(count)
+	updateLoop(mc.updateGauge, mc.interval)
+}
+
+func createHistogram(count int, mc metricCollector) {
+	mc.registerHistogram(count)
+	updateLoop(mc.updateHistogram, mc.interval)
+
+}
+
+func createSummary(count int, mc metricCollector) {
+	mc.registerSummary(count)
+	updateLoop(mc.updateSummary, mc.interval)
+}
+
+// createAll generates all 4 metric types
+// If isRandom is sent as true, createAll will generate randomized metrics. Otherwise createALl will steadily create the 4 types of metrics with a fixed count (provided by the user
+func createAll(count int, mc metricCollector, isRandom bool) {
+
+	if isRandom {
+		idx := rand.Intn(4)
+		lower := 1
+		upper := 4
+		amount := rand.Intn(upper-lower) + lower
+		metrics := []string{"counter", "gauge", "histogram", "summary"}
+		rands := []int{rand.Intn(200), rand.Intn(200), rand.Intn(200), rand.Intn(200)}
+		for i := 0; i <= amount; i++ {
+			if idx >= len(metrics) {
+				idx = 0
+			}
+			str := metrics[idx]
+			idx++
+			switch str {
+			case "counter":
+				createCounter(rands[0], mc)
+			case "gauge":
+				createGauge(rands[1], mc)
+			case "histogram":
+				createHistogram(rands[2], mc)
+			case "summary":
+				createSummary(rands[3], mc)
+			}
+		}
+
+	} else {
+		mc.registerCounter(count)
+		mc.registerGauge(count)
+		mc.registerHistogram(count)
+		mc.registerSummary(count)
+		go mc.updateMetrics()
+
+	}
+
+}
+
+// Run reads the config file and uses the data as default arguments.
+// These arguments can be overriden by CLI input (see README)
+func (cli *CommandLine) Run() {
+	data, err := ioutil.ReadFile("config.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+	var conf Config
+	if err := conf.Parse(data); err != nil {
+		log.Fatal(err)
+	}
+	generateCmd := flag.NewFlagSet("generate", flag.ExitOnError)
+
+	// Handling it without viper / cobra for now - still follows flags >  configuration file > defaults
+	// defaults are set first
+	// config file is read - if there are valid values, config file overrides defaults
+	// flags will use config values as default values and overide them with CLI input
+	usedType := defaultType
+	usedCount := defaultCount
+	usedFreq := defaultFreq
+	usedRand := defaultRand
+	usedAddress := defaultAddress
+	if conf.Type != "" {
+		usedType = conf.Type
+	}
+	if conf.Count >= 0 {
+		usedCount = conf.Count
+	}
+	if conf.Frequency >= 0 {
```

Comment:
```suggestion
	if conf.Count > 0 {
		usedCount = conf.Count
	}
	if conf.Frequency > 0 {
```

Using the zero values from `conf` probably doesn't make sense here.